test/functional/validation-functions-and-decorators.spec.ts

```diff
@@ -3719,6 +3726,387 @@ describe('IsUUID v5', () => {
+  const invalidValues = [
+    null,
+    undefined,
+    '',
+    'xxxA987FBC9-4BED-3078-CF07-9141BA07C9F3',
+    '934859',
+    'AAAAAAAA-1111-1111-AAAG-111111111111',
+    'A987FBC9-4BED-3078-8F07-9141BA07C9F3', // v3
+    '713ae7e3-cb32-45f9-adcb-7c4fa86b90c1', // v4
+  ];
+
+  class MyClass {
+    @IsUUID('8')
+    someProperty: string;
+  }
+
+  it('should not fail if validator.validate said that its valid', () => {
+    return checkValidValues(new MyClass(), validValues);
+  });
+
+  it('should fail if validator.validate said that its invalid', () => {
+    return checkInvalidValues(new MyClass(), invalidValues);
+  });
+
+  it('should not fail if method in validator said that its valid', () => {
+    validValues.forEach(value => expect(isUUID(value, '8')).toBeTruthy());
+  });
+
+  it('should fail if method in validator said that its invalid', () => {
+    invalidValues.forEach(value => expect(isUUID(value, '8')).toBeFalsy());
+  });
+
+  it('should return error object with proper data', () => {
+    const validationType = 'isUuid';
+    const message = 'someProperty must be a UUID';
+    return checkReturnedError(new MyClass(), invalidValues, validationType, message);
+  });
+});
+
+describe('IsUUID nil', () => {
+  const validValues = ['00000000-0000-0000-0000-000000000000'];
+  const invalidValues = [
+    null,
+    undefined,
+    '',
+    'xxxA987FBC9-4BED-3078-CF07-9141BA07C9F3',
+    '934859',
+    'AAAAAAAA-1111-1111-AAAG-111111111111',
+    'A987FBC9-4BED-3078-8F07-9141BA07C9F3', // v3
+    '713ae7e3-cb32-45f9-adcb-7c4fa86b90c1', // v4
+  ];
+
+  class MyClass {
+    @IsUUID('nil')
+    someProperty: string;
+  }
+
+  it('should not fail if validator.validate said that its valid', () => {
+    return checkValidValues(new MyClass(), validValues);
+  });
+
+  it('should fail if validator.validate said that its invalid', () => {
+    return checkInvalidValues(new MyClass(), invalidValues);
+  });
+
+  it('should not fail if method in validator said that its valid', () => {
+    validValues.forEach(value => expect(isUUID(value, 'nil')).toBeTruthy());
+  });
+
+  it('should fail if method in validator said that its invalid', () => {
+    invalidValues.forEach(value => expect(isUUID(value, 'nil')).toBeFalsy());
+  });
+
+  it('should return error object with proper data', () => {
+    const validationType = 'isUuid';
+    const message = 'someProperty must be a UUID';
+    return checkReturnedError(new MyClass(), invalidValues, validationType, message);
+  });
+});
+
+describe('IsUUID max', () => {
+  const validValues = ['ffffffff-ffff-ffff-ffff-ffffffffffff'];
+  const invalidValues = [
+    null,
+    undefined,
+    '',
+    'xxxA987FBC9-4BED-3078-CF07-9141BA07C9F3',
+    '934859',
+    'AAAAAAAA-1111-1111-AAAG-111111111111',
+    'A987FBC9-4BED-3078-8F07-9141BA07C9F3', // v3
+    '713ae7e3-cb32-45f9-adcb-7c4fa86b90c1', // v4
+  ];
+
+  class MyClass {
+    @IsUUID('max')
```

Comment:
Question: In which situation i would set constant validation to max uuid value? If i do that - its better to omit the field at all and put max value directly further in the code instead.